src/main/java/org/kuali/kfs/pdp/batch/service/impl/ExtractTypeContext.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * The Kuali Financial System, a comprehensive financial management system for higher education.
+ *
+ * Copyright 2005-2022 Kuali, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.kuali.kfs.pdp.batch.service.impl;
+
+import java.util.Date;
+
+import org.kuali.kfs.pdp.businessobject.PaymentProcess;
+import org.kuali.kfs.pdp.businessobject.PaymentStatus;
+
+import edu.cornell.kfs.pdp.batch.service.impl.PaymentUrgency;
+
+/**
+ * ====
+ * CU Customization: Added method for limiting extraction based on the payment's urgency.
+ * ====
+ * 
+ * The state (i.e. context) of the extraction being performed, which can differ depending on the
+ * {@link ExtractionType}.
+ */
+public interface ExtractTypeContext {
```

Comment:
Since we are adding a new function, would it be better to create CuPaymentUrgency that extends PaymentUrgency

Reply:
`PaymentUrgency` is a CU-specific enum that is being introduced by this PR. Did you mean to suggest creating `CuExtractTypeContext` that extends `ExtractTypeContext`?

I had set up the new function as a "default" one, so that implementing classes are not required to override/implement it. That makes it easier to introduce the new functionality without breaking existing code. (Also, due to the areas of the code that will be using this, it's much simpler to overlay the base interface than to introduce a CU-specific one.)

If you have other thoughts or recommendations on how to improve this, please let me know.

Reply:
I think that makes sense, lets go with this approach